### ISSUE TITLE
Tunnelblick is displaying it uses openvpn 2.4.4 as default, but use 2.3.18 instead in 3.7.5beta07

### ISSUE THREAD

**Issue by ohmer1** [Tunnelblick/Tunnelblick#436]
The VPN configuration is configured to use the default version of OpenVPN (2.4.4). With beta07, the client can't connect because of cipher not found. I rollbacked to beta05 and it works great again.

**Comment by jkbullard** [comment]
@ohmer1 - Were you using the same version of OpenVPN **and SSL**  on beta07 as in beta05? Perhaps you were using OpenSSL on one and LibreSSL on the other.

**Comment by ohmer1** [comment]
Same version for both.

beta05:
![beta05](https://framapic.org/Q0yPxlk7GLC5/1OAlkRS9o2Xy.png)

beta07:
![beta07](https://framapic.org/V4AWgZFylHRS/uhhwEqBiMEvB.png)

**Comment by ohmer1** [comment]
I can confirm that even if the version is set to default (2.4.4), Tunnelblick use 2.3.18-openssl.

  * Try to connect to a server
  * When Tunnelblick ask for passphrase, don't enter it,
  * Go to Terminal and type: ps aux | grep openvpn

**Comment by jkbullard** [comment]
In 3.7.5beta07, please "Reset Disabled Warnings" on the Preferences panel of Tunnelblick's "VPN Details" window. Then quit Tunnelblick and relaunch it and try to connect. If it still doesn't work and you don't get a warning explaining the problem, please post the diagnostic info from [Before You Post](https://tunnelblick.net/cBeforeYouPost.html).

**Comment by ohmer1** [comment]
No warnings displayed. When the first user reported the problem to me, I replicated the problem on my Mac and then I had to downgrade Tunnelblick on another user's computer. So, this is not specific to only one computer. I would probably never detected the problem if we were not using AES-256-GCM as cipher that is only supported by ovpn >=2.4.

```
*Tunnelblick: OS X 10.12.6; Tunnelblick 3.7.5beta07 (build 5002); prior version 3.7.5beta05 (build 4980); Standard user
git commit e762905b2b796eb2d1abcc21ccf505c7dcdf9d56


Configuration vpn-server1

"Sanitized" condensed configuration file for /Users/usertest/Library/Application Support/Tunnelblick/Configurations/vpn-server1.tblk:

remote (removed)
port 1193
proto udp
dev tun
resolv-retry infinite
client
nobind
persist-key
persist-tun
cipher AES-256-GCM
auth SHA256
tls-version-min 1.2
verb 2
mute 5
reneg-sec 0
ping 60
management-hold
management-query-passwords
auth-retry interact
remote-cert-tls server
<ca>
[Security-related line(s) omitted]
</ca>
<tls-auth>
[Security-related line(s) omitted]
</tls-auth>
<cert>
[Security-related line(s) omitted]
</cert>
<key>
[Security-related line(s) omitted]
</key>


================================================================================

Non-Apple kexts that are loaded:

Index Refs Address            Size       Wired      Name (Version) UUID <Linked Against>
  141    3 0xffffff7f83292000 0x63000    0x63000    org.virtualbox.kext.VBoxDrv (5.2.4) B81C8F96-8276-3708-938E-3BE8C8B48759 <7 5 4 3 1>
  143    0 0xffffff7f832f5000 0x8000     0x8000     org.virtualbox.kext.VBoxUSB (5.2.4) EAE7363A-28F0-3DDE-B6C0-7D1F2D50DC37 <142 141 40 7 5 4 3 1>
  144    0 0xffffff7f832fd000 0x5000     0x5000     org.virtualbox.kext.VBoxNetFlt (5.2.4) B375D15D-DD7E-363E-883A-8352873A1AD2 <141 7 5 4 3 1>
  145    0 0xffffff7f83302000 0x6000     0x6000     org.virtualbox.kext.VBoxNetAdp (5.2.4) EB116E49-1A55-3176-B29F-D91EEDD15BC9 <141 5 4 1>

================================================================================

There are no unusual files in vpn-server1.tblk

================================================================================

Configuration preferences:

-openvpnVersion = 
-notOKToCheckThatIPAddressDidNotChangeAfterConnection = 0
-lastConnectionSucceeded = 0

================================================================================

Wildcard preferences:

-notOKToCheckThatIPAddressDidNotChangeAfterConnection = 1

================================================================================

Program preferences:

launchAtNextLogin = 1
notOKToCheckThatIPAddressDidNotChangeAfterConnection = 1
tunnelblickVersionHistory = (
    "3.7.5beta07 (build 5002)",
    "3.7.5beta05 (build 4980)",
    "3.7.5beta07 (build 5002)",
    "3.7.5beta05 (build 4980)",
    "3.6.6 (build 4582)"
)
statusDisplayNumber = 0
lastLaunchTime = 540842495.791468
lastLanguageAtLaunchWasRTL = 0
connectionWindowDisplayCriteria = showWhenConnecting
maxLogDisplaySize = 102400
lastConnectedDisplayName = vpn-server1
keyboardShortcutIndex = 1
updateCheckAutomatically = 0
NSWindow Frame ConnectingWindow = 775 615 412 232 0 0 1920 1057 
NSWindow Frame SUUpdateAlert = 650 518 620 392 0 0 1920 1057 
detailsWindowFrameVersion = 5002
detailsWindowFrame = {{500, 461}, {920, 468}}
detailsWindowLeftFrame = {{0, 0}, {165, 350}}
detailsWindowViewIndex = 0
detailsWindowConfigurationsTabIdentifier = log
leftNavSelectedDisplayName = vpn-server1
AdvancedWindowTabIdentifier = connectingAndDisconnecting
haveDealtWithOldTunTapPreferences = 1
haveDealtWithOldLoginItem = 1
haveDealtWithAfterDisconnect = 1
SUEnableAutomaticChecks = 0
SUFeedURL = https://www.tunnelblick.net/appcast-s.rss
SUScheduledCheckInterval = 86400
SUSendProfileInfo = 0
SULastCheckTime = 2018-02-20 14:38:18 +0000
SUHasLaunchedBefore = 1
WebKitDefaultFontSize = 11
WebKitStandardFont = .AppleSystemUIFont
askedUserIfOKToCheckThatIPAddressDidNotChangeAfterConnection = 1
haveDealtWithSparkle1dot5b6 = 1
updateSendProfileInfo = 1

================================================================================

Tunnelblick Log:

*Tunnelblick: OS X 10.12.6; Tunnelblick 3.7.5beta07 (build 5002); prior version 3.7.5beta05 (build 4980)
2018-02-20 13:04:19 *Tunnelblick: Attempting connection with vpn-server1 using shadow copy; Set nameserver = 769; monitoring connection
2018-02-20 13:04:19 *Tunnelblick: openvpnstart start vpn-server1.tblk 1337 769 0 1 0 1065264 -ptADGNWradsgnw 2.3.18-openssl-1.0.2n
2018-02-20 13:04:20 *Tunnelblick: openvpnstart log:
     OpenVPN started successfully. Command used to start OpenVPN (one argument per displayed line):
     
          /Applications/Tunnelblick.app/Contents/Resources/openvpn/openvpn-2.3.18-openssl-1.0.2n/openvpn
          --daemon
          --log
          /Library/Application Support/Tunnelblick/Logs/-SUsers-Susertest-SLibrary-SApplication Support-STunnelblick-SConfigurations-Svpn--server1.tblk-SContents-SResources-Sconfig.ovpn.769_0_1_0_1065264.1337.openvpn.log
          --cd
          /Library/Application Support/Tunnelblick/Users/usertest/vpn-server1.tblk/Contents/Resources
          --setenv
          IV_GUI_VER
          "net.tunnelblick.tunnelblick 5002 3.7.5beta07 (build 5002)"
          --verb
          3
          --config
          /Library/Application Support/Tunnelblick/Users/usertest/vpn-server1.tblk/Contents/Resources/config.ovpn
          --verb
          3
          --cd
          /Library/Application Support/Tunnelblick/Users/usertest/vpn-server1.tblk/Contents/Resources
          --management
          127.0.0.1
          1337
          --management-query-passwords
          --management-hold
          --script-security
          2
          --up
          /Applications/Tunnelblick.app/Contents/Resources/client.up.tunnelblick.sh -9 -d -f -m -w -ptADGNWradsgnw
          --down
          /Applications/Tunnelblick.app/Contents/Resources/client.down.tunnelblick.sh -9 -d -f -m -w -ptADGNWradsgnw

2018-02-20 13:04:19 *Tunnelblick: openvpnstart starting OpenVPN
2018-02-20 13:04:20 OpenVPN 2.3.18 x86_64-apple-darwin [SSL (OpenSSL)] [LZO] [PKCS11] [MH] [IPv6] built on Feb 16 2018
2018-02-20 13:04:20 library versions: OpenSSL 1.0.2n  7 Dec 2017, LZO 2.10
2018-02-20 13:04:20 MANAGEMENT: TCP Socket listening on [AF_INET]127.0.0.1:1337
2018-02-20 13:04:20 Need hold release from management interface, waiting...
2018-02-20 13:04:20 MANAGEMENT: Client connected from [AF_INET]127.0.0.1:1337
2018-02-20 13:04:20 *Tunnelblick: Established communication with OpenVPN
2018-02-20 13:04:20 MANAGEMENT: CMD 'pid'
2018-02-20 13:04:20 MANAGEMENT: CMD 'state on'
2018-02-20 13:04:20 MANAGEMENT: CMD 'state'
2018-02-20 13:04:20 MANAGEMENT: CMD 'bytecount 1'
2018-02-20 13:04:20 MANAGEMENT: CMD 'hold release'
2018-02-20 13:04:20 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
2018-02-20 13:04:23 MANAGEMENT: CMD 'password [...]'
2018-02-20 13:04:23 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
2018-02-20 13:04:23 MANAGEMENT: Client disconnected
2018-02-20 13:04:23 Cipher algorithm 'AES-256-GCM' not found
2018-02-20 13:04:23 Exiting due to fatal error
2018-02-20 13:04:24 *Tunnelblick: No 'post-disconnect.sh' script to execute
2018-02-20 13:04:24 *Tunnelblick: Expected disconnection occurred.

================================================================================

"Sanitized" full configuration file

remote (removed)
port 1193
proto udp
dev tun
resolv-retry infinite
client
nobind
persist-key
persist-tun
cipher AES-256-GCM
auth SHA256
tls-version-min 1.2
verb 2
mute 5
reneg-sec 0
ping 60
#management 127.0.0.1 31192
management-hold
management-query-passwords
auth-retry interact
remote-cert-tls server

<ca>
 [Security-related line(s) omitted]
</ca>

<tls-auth>
 [Security-related line(s) omitted]
</tls-auth>


<cert>
 [Security-related line(s) omitted]
</cert>
 
<key>
 [Security-related line(s) omitted]
</key>



================================================================================

ifconfig output:

lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 16384
	options=1203<RXCSUM,TXCSUM,TXSTATUS,SW_TIMESTAMP>
	inet 127.0.0.1 netmask 0xff000000 
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
	nd6 options=201<PERFORMNUD,DAD>
gif0: flags=8010<POINTOPOINT,MULTICAST> mtu 1280
stf0: flags=0<> mtu 1280
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=10b<RXCSUM,TXCSUM,VLAN_HWTAGGING,AV>
	ether (removed) 
	inet (removed) netmask 0xfffff800 broadcast (removed)
	inet6 fe80::40e:2632:d150:2495%en0 prefixlen 64 secured scopeid 0x4 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect (1000baseT <full-duplex>)
	status: active
en1: flags=8823<UP,BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	ether (removed)
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect (<unknown type>)
	status: inactive
en2: flags=963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX> mtu 1500
	options=60<TSO4,TSO6>
	ether (removed) 
	media: autoselect <full-duplex>
	status: inactive
en3: flags=963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX> mtu 1500
	options=60<TSO4,TSO6>
	ether (removed)
	media: autoselect <full-duplex>
	status: inactive
bridge0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=63<RXCSUM,TXCSUM,TSO4,TSO6>
	ether (removed)
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x2
	member: en2 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 6 priority 0 path cost 0
	member: en3 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 7 priority 0 path cost 0
	nd6 options=201<PERFORMNUD,DAD>
	media: <unknown type>
	status: inactive
p2p0: flags=8802<BROADCAST,SIMPLEX,MULTICAST> mtu 2304
	ether(removed)
	media: autoselect
	status: inactive
awdl0: flags=8902<BROADCAST,PROMISC,SIMPLEX,MULTICAST> mtu 1484
	ether (removed)
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: inactive
utun0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 2000
	inet6 fe80::(removed)%utun0 prefixlen 64 scopeid 0xb 
	nd6 options=201<PERFORMNUD,DAD>
utun1: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 2000
	inet6 fe80::(removed)%utun1 prefixlen 64 scopeid 0xc 
	nd6 options=201<PERFORMNUD,DAD>

================================================================================

Console Log:

2018-02-20 13:00:48 Tunnelblick[1043] applicationShouldTerminate: termination because of Quit; delayed until 'shutdownTunnelblick' finishes
2018-02-20 13:00:48 Tunnelblick[1043] Finished shutting down Tunnelblick; allowing termination
2018-02-20 13:00:52 Tunnelblick[10773] Tunnelblick: OS X 10.12.6; Tunnelblick 3.7.5beta07 (build 5002)
2018-02-20 13:01:33 Tunnelblick[10773] applicationShouldTerminate: termination because of Quit; delayed until 'shutdownTunnelblick' finishes
2018-02-20 13:01:33 Tunnelblick[10773] Finished shutting down Tunnelblick; allowing termination
2018-02-20 13:01:35 Tunnelblick[10798] Tunnelblick: OS X 10.12.6; Tunnelblick 3.7.5beta07 (build 5002)
```

**Comment by jkbullard** [comment]
@ohmer1 - Thank you very much. Your new title for this describes the problem perfectly.

This problem is fixed in 3.7.5beta08, which was released a few minutes ago.
